File: services/services/Together_ai/together ai.ts
import OpenAI from "openai";

const together = new OpenAI({
  apiKey: process.env.TOGETHER_API_KEY,
  baseURL: "https://api.together.xyz/v1",
});

export async function togetherSentiment(prompt: string) {
  const completion = await together.chat.completions.create({
    model: "meta-llama/Meta-Llama-3.1-70B-Instruct",
    messages: [{ role: "user", content: prompt }],
  });

  return completion.choices[0].message.content;
}

File: services/services/Together_ai/together ai.py
def stream_multi_model(symbol, news):
    models = [
        "deepseek-ai/DeepSeek-R1",              # reasoning
        "meta-llama/Meta-Llama-3.1-70B-Instruct"  # general sentiment
    ]

    for model in models:
        stream = client.chat.completions.create(
            model=model,
            messages=[{"role": "user", "content": f"{symbol} news:\n{news}"}],
            stream=True
        )
        for chunk in stream:
            token = chunk.choices[0].delta.get("content", "")
            yield f"[{model}] {token}"
